webapp/lib/controller.dart

```diff
@@ -1448,6 +1445,9 @@ void showAndLogError(error, trace) {
     errMsg = "A network problem occurred: ${error.message}";
   } else if (error is Exception) {
     errMsg = "An internal error occurred: ${error.runtimeType}";
+  } else if (error is FirebaseError) {
```

Comment:
Since `FirebaseError` is an abstract class, then it may now or in the future be a subclass of `Exception`. If that is or ever becomes the case, then this `else if` clause will never be triggered because the `else if` clause above it will be true. To address this, I recommend putting the `else if (error is FirebaseError) {` clause before the `else if (error is Exception) {` clause

Reply:
Good catch. Refactored it.